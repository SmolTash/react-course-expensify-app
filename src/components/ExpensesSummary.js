import React from 'react';
import {connect} from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses' ;
    const formattedTotal = numeral( expensesTotal/100).format('$0,0.00');
    return (
        <div> 
        <p> 
            Viewing  {expensesCount } {expenseWord} totaling  {formattedTotal} 
        </p>       
    </div>
    );   
};

const mapStateToProps =  (state) => {
    const  expenses = selectExpenses (state.expenses, state.filters)    
    return {
        expensesCount: expenses.length,          
        expensesTotal: selectExpensesTotal (expenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);
