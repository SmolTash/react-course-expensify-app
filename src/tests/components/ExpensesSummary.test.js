import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test ('hould correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow (<ExpensesSummary expensesCount={1} expensesTotal = {256} />);
    expect(wrapper).toMatchSnapshot();
});

test ('hould correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow (<ExpensesSummary expensesCount={13} expensesTotal = {789256} />);
    expect(wrapper).toMatchSnapshot();
}); 