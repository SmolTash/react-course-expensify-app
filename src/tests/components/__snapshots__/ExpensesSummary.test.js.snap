// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hould correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing  
    1
     
    expense
     totaling  
    $2.56
  </p>
</div>
`;

exports[`hould correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing  
    13
     
    expenses
     totaling  
    $7,892.56
  </p>
</div>
`;
